core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -285,21 +297,27 @@ private void tryStart(Profiler profiler) {
     /**
      * Stops the container.
      */
+    @Override
     public void stop() {
 
         if (containerId == null) {
             return;
         }
 
-        String imageName;
-
         try {
-            imageName = image.get();
-        } catch (Exception e) {
-            imageName = "<unknown>";
-        }
+            String imageName;
+
+            try {
+                imageName = image.get();
+            } catch (Exception e) {
+                imageName = "<unknown>";
```

Comment:
I see it's just copied from the old implementation, but does this acutally happen?

Reply:
well, to give you a bit of history here:

When I was implementing https://github.com/testcontainers/testcontainers-java/pull/106/files, we wanted to keep the backward compatibility as much as possible, and one of such is Logger based on image's name :)